src/main/io/gps.c

```diff
@@ -1523,6 +1972,11 @@ static union {
     ubxNavSat_t sat;
     ubxCfgGnss_t gnss;
     ubxAck_t ack;
+    struct {
+        char swVersion[30];
+        char hwVersion[10];
+        char extension[300];
```

Comment:
```suggestion
        char extension[10][30];
```
It would be easier to read and get right (generated code is equivalent)

Reply:
From what I understand, this field may hold fewer than 10 iterations and the content may be less than 30 bytes; the documentation is unclear.
At present we don't parse the field, so I'll make a comment and leave it as is.

Reply:
The UBX docs don't say what the max size of the extensions can be.

However, they list and describe nine different fields that could be found in the extension, so we could infer the max fields is nine. I have never seen more than five in the 26 different modules I have, but still a small sample size.

Each extension field is a fixed 30 bytes. They are null terminated strings, and any unused portion of the 30 bytes is undefined byte values, but I've always seen them be null filled.

Yes, expressing it as `extension[10][30]` would more clearly communicate the structure of the extension field. We could make the '10' a define like `MAX_SUPPORTED_EXT_FIELDS` or some such (like we do with max sats, to more clearly control this union element size.

We could probably at least reduce the size by one field to 9 * 30 = 270.

We could also treat this like the sat info, where we size it to something less, parse the lesser number of fields and toss any extras returned. I've never seen more than 5.

As ctzsnooze mentioned, we don't use any info from here right now. What we might someday want out of extensions is the PROTVER (protocol version), and maybe the supported constellations list. But that's pure speculation, we have no use for them today.

Although it seems like the extensions are always ordered with FWVER and PROTVER first, that's not guaranteed by the spec, so if we reduce the fields parsed we may miss the ones we want.

Similar to sat info, however, if UBX-MON-VER returns more fields than we parse, we still need to calculate the checksum off the entire returned contents to avoid tossing the entire message.

Reply:
@ZzyzxTek : `extension` is not a problem in `_buffer`, as long as it is not longest member. My original problem was with `extension` stored in memory for CLI.
Ideally, parsing can be implemented to process  all 30byte chunks that fit in buffer( = min(received size, sizeof(_buffer) ). But parsed values shall be memory efficient (parsed version is only few bytes, just store FWVER and PROTVER)

Then we can even define `extension[0][30]`